src/SFML/Window/OSX/JoystickImpl.cpp

```diff
@@ -356,8 +362,8 @@ JoystickCaps JoystickImpl::getCapabilities() const
     caps.buttonCount = m_buttons.size();
 
     // Axis:
-    for (AxisMap::const_iterator it(m_axis.begin()); it != m_axis.end(); ++it)
-        caps.axes[it->first] = true;
+    for (const auto& [axis, iohidElementRef] : m_axis)
```

Comment:
There's no way to ignore a destructuring binding (here `iohidElementRef`), but doesn't this cause unused-variable warnings?

Unfortunately, `std::ignore` only works with `std::tie()`, which cannot be used for initialization. ([some SO discussion](https://stackoverflow.com/q/40673080))

Reply:
There's really no way to ignore a single binding in a destructed binding declaration. This is a (known) annoying limitation.

Reply:
There was a [SO comment](https://stackoverflow.com/questions/40673080/stdignore-with-structured-bindings#comment76711644_40680014) saying it would generate a warning, but maybe that's not accurate or no longer the case.

I'm happy if this passes CI, no need to create problems where there are none 🙂

Reply:
CI seems happy, let's merge!

Reply:
@Bromeon ping :)